docs/directory_structure.md

```diff
@@ -0,0 +1,99 @@
+## Directory Structure
```

Comment:
Can we replace it with https://github.com/Cloud-CV/EvalAI/wiki/Directory-Structure? Minor styling and grammatical fixes.